### PR TITLE
Add brand-fetch extension

### PR DESCRIPTION
**Pull request by dilutedev**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/brand-fetch/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "brand-fetch",
+  "title": "brand-fetch",
```
Comment (pernielsentikaer) [inline extensions/brand-fetch/package.json:4]:
```suggestion
  "title": "Brandfetch",
```

File: extensions/brand-fetch/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# brand-fetch Changelog
```
Comment (pernielsentikaer) [inline extensions/brand-fetch/CHANGELOG.md:1]:
```suggestion
# Brandfetch Changelog
```

File: extensions/brand-fetch/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# brand-fetch
```
Comment (pernielsentikaer) [inline extensions/brand-fetch/README.md:1]:
```suggestion
# Brandfetch
```

File: extensions/brand-fetch/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+      isLoading={isLoading}
+    >
+      <Form.Description text="This form accepts fields required to interact with brand fetch" />
+      <Form.TextField id="textfield" title="Brand Domain" placeholder="Enter text" defaultValue="brandfetch.com" />
```
Comment (pernielsentikaer) [inline extensions/brand-fetch/src/index.tsx:77]: What else could be entered here?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- How does it work, does users have multiple domains? We might look into a better UX

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/brand-fetch/src/index.tsx (hunk 4 above)
Comment (dilutedev) [reply extensions/brand-fetch/src/index.tsx:77]: this is pretty much all the info required for this api

**Comment by dilutedev** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * How does it work, does users have multiple domains? We might look into a better UX
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

This is an integration with [the](https://docs.brandfetch.com/reference/retrieve-brand) brandfetch api. Users can send multiple domain names and fetch logos in various formats

File: extensions/brand-fetch/src/index.tsx (hunk 4 above)
Comment (pernielsentikaer) [reply extensions/brand-fetch/src/index.tsx:77]: So the user would just enter one domain, right? Would it make sense to land on a list instead where you can select the domain? Could you maybe record a simple screencast of the flow so I can get a better understanding 🙂

Comment (dilutedev) [reply extensions/brand-fetch/src/index.tsx:77]: yes just one domain.
Theoritically, a user would be able to input any domain from all over the web into this extension, so we cant have this option as a list. Does that make any sense?

**Comment by pernielsentikaer** [comment]
DM'ed you in Slack 🙂

**Comment by pernielsentikaer** [comment]
Are you still around @dilutedev

**Comment by dilutedev** [comment]
Yes, I will deploy those requested updates by eod today. Is that ok?

**Comment by pernielsentikaer** [comment]
Is this ready for another review or not yet?

**Comment by dilutedev** [comment]
> Is this ready for another review or not yet?

yes it is

**Comment by pernielsentikaer** [comment]
What do you think about #11680 - do you agree it makes senes to keep it as two different extensions?

**Comment by dilutedev** [comment]
> What do you think about #11680 - do you agree it makes senes to keep it as two different extensions?

I think if possible we could merge into one extension, this way a user needs to install only one extension for every use-case wrt brandfetch.

**Comment by pernielsentikaer** [comment]
But one requires credentials and one doesn't - if you only want to search for icons without credentials it's not possible

**Comment by dilutedev** [comment]
> But one requires credentials and one doesn't - if you only want to search for icons without credentials it's not possible

True. We keep them as seperate extensions then.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**